core/interface.py

```diff
@@ -64,11 +64,17 @@ def __get__(self, obj=None, cls=None):
         return self
 
     def __call__(self, *args, **kwargs):
+        if self.__isabstractmethod__:
+            raise NotImplementedError('No methods registered on abstractmethod {0}.'
+                                      ''.format(self.default_callable.__name__))
+        elif self.registered:
```

Comment:
you are not checking the args or kwargs, how can you than determine that no keyword was given?